src/board/helpers.mjs

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable guard-for-in */
+/* eslint-disable no-restricted-syntax */
+
+// contains functions to duplicate, flatten, unflatten, and shuffle the grid
+
+function duplicateGrid(grid) {
+  const result = [];
+  grid.forEach((row, gridIndex) => {
+    const newRow = [];
+    row.forEach((color, rowIndex) => {
+      newRow[rowIndex] = color;
+    });
+    result[gridIndex] = newRow;
+  });
+  return result;
+}
+
+function flattenGrid(grid) {
+  const result = [];
+  grid.forEach((row) => {
+    row.forEach((color) => {
+      result.push(color);
+    });
+  });
+  return result;
+}
+
+function shuffle(arr) { // Fisher-Yates shuffle
+  for (let i = arr.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1));
+    // TODO: ask about this
+    // eslint-disable-next-line no-param-reassign
+    [arr[i], arr[j]] = [arr[j], arr[i]];
+  }
+  return arr;
+}
+
+function flatToMultiDimensional(flatGrid) {
+  // square root of the size of the grid will tell us how many rows/columns
+  const rowSize = Math.sqrt(flatGrid.length);
+
+  // create the empty multidimensional grid
+  const multiDimensionalGrid = new Array(rowSize);
+
+  // why can I not turn this into a forEach or map
```

Comment:
thats a good question, i would assume you be able to based on the code, maybe we could look into this